diff -ruN com_github_golangci_gofmt/gofmt/BUILD.bazel com_github_golangci_gofmt_b/gofmt/BUILD.bazel
--- b/gofmt/BUILD.bazel	2020-08-26 17:26:46.000000000 +0200
+++ c/gofmt/BUILD.bazel	2020-08-26 22:52:44.000000000 +0200
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -14,6 +14,20 @@
     visibility = ["//visibility:public"],
 )
 
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "doc.go",
+        "gofmt.go",
+        "golangci.go",
+        "internal.go",
+        "rewrite.go",
+        "simplify.go",
+    ],
+    importpath = "github.com/golangci/gofmt/gofmt",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = [

--- b/goimports/BUILD.bazel	2020-09-04 12:48:16.000000000 +0200
+++ c/goimports/BUILD.bazel	2020-09-04 12:49:19.000000000 +0200
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_tool_library")
 
 go_library(
     name = "go_default_library",
@@ -10,3 +10,14 @@
     visibility = ["//visibility:public"],
     deps = ["@org_golang_x_tools//imports:go_default_library"],
 )
+
+go_tool_library(
+    name = "go_tool_library",
+    srcs = [
+        "goimports.go",
+        "golangci.go",
+    ],
+    importpath = "github.com/golangci/gofmt/goimports",
+    visibility = ["//visibility:public"],
+    deps = ["@org_golang_x_tools//imports:go_tool_library"],
+)
